autoscaler/controllers/gateway/configmap.go

```diff
@@ -23,6 +27,87 @@ const (
 	destinationConfiguredType = "DestinationConfigured"
 )
 
+var (
+	errNoPipelineConfigured = errors.New("no pipeline was configured, cannot add self telemetry pipeline")
+	errNoReceiversConfigured = errors.New("no receivers were configured, cannot add self telemetry pipeline")
+	errNoExportersConfigured = errors.New("no exporters were configured, cannot add self telemetry pipeline")
+)
+
+func addSelfTelemetryPipeline(c *config.Config) error {
+	if c.Service.Pipelines == nil {
+		return errNoPipelineConfigured
+	}
+	if c.Receivers == nil {
+		return errNoReceiversConfigured
+	}
+	if c.Exporters == nil {
+		return errNoExportersConfigured
+	}
+	c.Receivers["prometheus"] = config.GenericMap{
+		"config": config.GenericMap{
+			"scrape_configs": []config.GenericMap{
+				{
+					"job_name": "otelcol",
+					"scrape_interval": "10s",
+					"static_configs": []config.GenericMap{
+						{
+							"targets": []string{"127.0.0.1:8888"},
+						},
+					},
+					"metric_relabel_configs": []config.GenericMap{
+						{
+							"source_labels": []string{"__name__"},
+							"regex": "(.*odigos.*|^otelcol_processor_accepted.*|^otelcol_exporter_sent.*)",
+							"action": "keep",
+						},
+					},
+				},
+			},
+		},
+	}
+	if c.Processors == nil {
+		c.Processors = make(config.GenericMap)
+	}
+	c.Processors["resource/pod-name"] = config.GenericMap{
+		"attributes": []config.GenericMap{
+			{
+				"key":    "k8s.pod.name",
+				"value":  "${POD_NAME}",
+				"action": "upsert",
+			},
+		},
+	}
+	// odigostrafficmetrics processor should be the last processor in the pipeline
+	// as it helps to calculate the size of the data being exported.
+	// In case of performance impact caused by this processor, we should modify this config to reduce the sampling ratio.
+	c.Processors["odigostrafficmetrics"] = struct{}{}
+	c.Exporters["otlp/ui"] = config.GenericMap{
```

Comment:
Same retry/timeout configuration as in data-collection

Reply:
disabled retry